lib/mail/constants.rb

```diff
@@ -35,8 +35,27 @@ module Constants
     ATOM_UNSAFE   = /[#{Regexp.quote aspecial}#{control}#{sp}]/n
     PHRASE_UNSAFE = /[#{Regexp.quote aspecial}#{control}]/n
     TOKEN_UNSAFE  = /[#{Regexp.quote tspecial}#{control}#{sp}]/n
-    ENCODED_VALUE = /\=\?([^?]+)\?([QB])\?[^?]*?\?+\=/mi
-    FULL_ENCODED_VALUE = /(\=\?[^?]+\?[QB]\?[^?]*?\?+\=)/mi
+
+    ENCODED_VALUE = %r{
+      \=\?     # literal =?
+      ([^?]+)  #
+      \?([QB]) # either a "Q" or a "B"
+      \?       # literal ?
+      [^?]?.*? #
```

Comment:
This is only part that's changed. Before `[^?]*?` after `[^?]?.*?`.

I must admit I was "inspired" by the python implementation here: https://github.com/python/cpython/blob/main/Lib/email/header.py#L41

Reply:
Can we ditch the `[^?]?` portion? The `.*?` would capture that as well.

Reply:
Ah, yes! Good point. Done.